Library/Homebrew/cmd/test-bot.rb

```diff
@@ -1017,7 +1018,14 @@ def test_bot
       end
     end
   ensure
-    HOMEBREW_CACHE.children.each(&:rmtree) if ARGV.include? "--clean-cache"
+    if ARGV.include? "--clean-cache"
+      HOMEBREW_CACHE.children.each(&:rmtree)
+    else
+      Dir.glob("*.bottle*.tar.gz") do |bottle_file|
+        FileUtils.rm_f HOMEBREW_CACHE/bottle_file
+      end
+    end
```

Comment:
This will only delete new generated bottles. Noticed that it's scanning on current directory not HOMEBREW_CACHE.